### PR TITLE
Fix regex for redhat

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

File: src/PlatformNames.jl [hunk 1]
```diff
@@ -155,19 +155,19 @@ function platform_key(machine::AbstractString = Sys.MACHINE)
     if startswith(machine, "x86_64-apple-darwin")
         return MacOS()
     end
-    if ismatch(r"x86_64-(pc-)?(unknown-)?linux-gnu", machine)
+    if ismatch(r"x86_64-(pc-)?(unknown-)?(redhat-)?linux(-gnu)?", machine)
```
Comment (ararslan) [inline src/PlatformNames.jl]: Is `x86_64-unknown-redhat-linux` valid? If not, the `unknown` might need to be `(unknown-|redhat-)?`.

Comment (ChrisRackauckas) [reply src/PlatformNames.jl]: Not sure. Someone else would have to chime in.

**Comment by staticfloat** [comment]
Honestly, we don't care if it's `redhat`, `unknown` or whatever.  Let's just make it `"x86_64-.*-linux(-gnu)?"`.  This way we continue to match only `x86_64` machines that use `glibc` on linux, which is what we care about.  (`musl` targets must use `-linux-musl*` at the end)

We should do the same for 32-bit as well of course.

File: src/PlatformNames.jl [hunk 2]
```diff
@@ -155,19 +155,20 @@ function platform_key(machine::AbstractString = Sys.MACHINE)
     if startswith(machine, "x86_64-apple-darwin")
         return MacOS()
     end
-    if ismatch(r"x86_64-(pc-)?(unknown-)?(redhat-)?linux(-gnu)?", machine)
+    if ismatch(r"x86_64-.*-linux(-gnu)?", machine)
         return Linux(:x86_64)
     end
-    if ismatch(r"i\d86-(pc-)?(unknown-)?(redhat-)?linux(-gnu)?", machine)
+       x86_64-.*-linux(-gnu)?
```
Comment (staticfloat) [inline src/PlatformNames.jl]: I'm guessing this is a typo

Comment (ChrisRackauckas) [reply src/PlatformNames.jl]: Yupyup

**Pull request merged**